src/allmydata/storage/http_client.py

```diff
@@ -475,28 +479,32 @@ class StorageClientGeneral(object):
 
     _client: StorageClient
 
-    @inlineCallbacks
-    def get_version(self):
+    @async_to_deferred
+    async def get_version(self):
         """
         Return the version metadata for the server.
         """
         url = self._client.relative_url("/storage/v1/version")
-        response = yield self._client.request("GET", url)
-        decoded_response = yield self._client.decode_cbor(
-            response, _SCHEMAS["get_version"]
+        response = await self._client.request("GET", url)
+        decoded_response = cast(
```

Comment:
I'm not entirely sure how to interpret this cast and the one below, mostly since this is an unchecked context so mypy doesn't care either way.

I almost commented that `# type: ignore[...]` might be preferable here but then I checked and mypy has options for warning about unused ignores _and_ redundant casts, so I guess it doesn't matter, as long as we write the write mypy config.

So nothing actionable here I guess.

Reply:
Pretty sure I fixed this because was getting type errors.

Reply:
Possibly just in pyright, though? So mypy would eventually start complaining too.